components/camera/replaypcd/replaypcd.go

```diff
@@ -135,38 +151,110 @@ func (replay *pcdCamera) NextPointCloud(ctx context.Context) (pointcloud.PointCl
 			SortOrder: datapb.Order_ORDER_ASCENDING,
 		},
 		CountOnly:     false,
-		IncludeBinary: true,
+		IncludeBinary: replay.limit == 1,
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	// If no data is returned, return an error indicating we've reached the end of the dataset.
-	if len(resp.GetData()) == 0 {
-		return nil, errEndOfDataset
-	}
-
 	replay.lastData = resp.GetLast()
-	data := resp.Data[0].GetBinary()
-
-	r, err := gzip.NewReader(bytes.NewBuffer(data))
-	if err != nil {
-		return nil, errors.Errorf("Failed to initialize gzip reader: %v", err)
-	}
 
-	defer func() {
-		if err = r.Close(); err != nil {
-			replay.logger.Warnw("Failed to close gzip reader", "warn", err)
+	// If batching use metadata to download data in parallel, caching results
+	if replay.limit != 1 {
+		for _, dataResponse := range resp.Data {
+			emptyCache := cachedData{id: dataResponse.GetMetadata().Id}
+			replay.cachedData = append(replay.cachedData, emptyCache)
 		}
-	}()
+		ctxTimeout, cancelTimeout := context.WithTimeout(ctx, time.Second)
```

Comment:
Timeout was decided arbitrarily, I had not thought of larger datasets but I agree for now its better to let performance lapse than erroring out too soon. This can be tuned in the future but we can start with a 30s timeout

Reply:
added a const at the top for this

Reply:
ty!